File: leetcode/Math/0202_Happy_Number.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        num_set = set()
        num_set.add(n)

        while True:
            sum = 0
            while n != 0:
                sum += (n % 10) * (n % 10)
                n //= 10
            
            if sum == 1:
                return True
            elif sum in num_set:
                return False
            else:
                num_set.add(sum)
                n = sum

File: leetcode/Math/0202_Happy_Number.cpp
#include <math.h>
class Solution {
public:
    bool isHappy(int n) {
        // 1 ~ 10, only 1 and y are happy number
        while (true) {
            int temp = 0;
            while (n != 0) {
                temp += pow(n % 10, 2);
                n /= 10;
            }
            if (temp == 1 || temp == 7)
                return true;
            else if (temp < 10)
                return false;
            n = temp;
        }
    }
};